rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,386 @@
+import gym
+import pytest
+import numpy as np
+import torch
+from typing import Any, List, Mapping, Union
+import unittest
+
+from ray.rllib.algorithms import AlgorithmConfig
+from ray.rllib.offline import IOContext
+from ray.rllib.offline.dataset_reader import (
+    DatasetReader,
+    get_dataset_and_shards,
+)
+
+import ray
+from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import FCNet, FCConfig
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.torch.torch_rl_module import TorchRLModule
+from ray.rllib.core.rl_module.torch.tests.test_torch_rl_module import (
+    to_numpy,
+    to_tensor,
+)
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.models.specs.specs_dict import ModelSpec
+from ray.rllib.models.specs.specs_torch import TorchTensorSpec
+from ray.rllib.models.torch.torch_distributions import (
+    TorchDeterministic,
+    TorchCategorical,
+)
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.test_utils import check
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+from ray.rllib.utils.typing import TensorType
+
+
+def model_norm(model: torch.nn.Module) -> float:
+    """Compute the norm of the weights of a model.
+
+    Args:
+        model: The model whose weights to compute a norm over.
+
+    Returns:
+        the norm of model.
+    """
+    total_norm = 0
+    for p in model.parameters():
+        param_norm = p.detach().data.norm(2)
+        total_norm += param_norm.item() ** 2
+    total_norm = total_norm**0.5
+    return total_norm
+
+
+def do_rollouts(
+    env: gym.Env, module_for_inference: RLModule, num_rollouts: int
+) -> SampleBatch:
+    _returns = []
```

Comment:
nit: it's actually kinda weird to have local variables in a function to have _ prefix.
these are not accessible once you go out of the function, so there is no need to specify accessibility?